### DISCUSSION TITLE
Adding linked lists to each other

### DISCUSSION THREAD

**Discussion by ABehniwal** [dotnet/runtime#108149] [Q&A] [answered]
https://sharplab.io/#v2:C4LgTgrgdgNAJiA1AHwAICYAMBYAUBgRj1UwAJUCAWAbjzwBkBLKAawFM4mBnYAHmeAA+UsDY8yAXlJQ2Ad1JNWHbnwGCAFAEpauBs3adGPflCEixwAqSkz5igypNCtOvKPEA6AIJw4AMUYwHnUCbTcLTG9fAKDgdXQw3HdgSJ9/QOCAZkTk1OiMuMpE8J4CKPTY9QBWHIsytJjggDZa0vLGuIB2Vst2gvUADmLcADMAezA2AEMAYwALUnUANymwUhZSZnNxTVIAbzxSI/ICAE51FkTj0gBfOnwz9QAiZKsoMdkn4fHJ2YXl1brTZQbaWXYHXDXCjnS46a53XQPc6vDwdYYo+hTHgeABybAAHsBrKDIh1qEA

In the above code, I have two linked lists and I want to add the second linked list to the end of my first linked list. However, the Next and Previous pointers of nodes aren't modifiable in the linked list template. Is there a way I can do this without having to iterate through the linked list and add each element individually.

**Comment by EgorBo** [comment, marked as answer]
> Is there a way I can do this without having to iterate through the linked list and add each element individually.

It's not possible to have an O(1) insert of one LinkedList into another without side-effects. E.g. if you add list1 into list2 without iterating, then if you remove an element from list2 it will be, effectively, removed from list1 as well.

Linked List is not a complicated data structure, if you need a fast merge (and are fine with the side-effects) you can roll your own fairly quickly.

**Reply by ABehniwal** [reply]
Okay, thank you!
